File: backend/models/KhuyenMai.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const KhuyenMai = sequelize.define(
  "KhuyenMai",
  {
    id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true },
    ma_khuyen_mai: {
      type: DataTypes.STRING(50),
      unique: true,
      allowNull: false,
    },
    ten_chuong_trinh: { type: DataTypes.STRING(150) },
    loai: { type: DataTypes.STRING(20) },
    gia_tri: { type: DataTypes.DECIMAL(15, 2), allowNull: false },
    gia_tri_toi_da: { type: DataTypes.DECIMAL(15, 2) },
    don_hang_toi_thieu: { type: DataTypes.DECIMAL(15, 2) },
    so_luong_ma: { type: DataTypes.INTEGER },
    da_su_dung: { type: DataTypes.INTEGER, defaultValue: 0 },
    ngay_bat_dau: { type: DataTypes.DATE },
    ngay_ket_thuc: { type: DataTypes.DATE },
    trang_thai: { type: DataTypes.STRING(20), defaultValue: "active" },
  },
  { tableName: "KhuyenMai", timestamps: false },
);

module.exports = KhuyenMai;

File: backend-csharp/Models/KhuyenMai.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class KhuyenMai
{
    public long Id { get; set; }

    public string MaKhuyenMai { get; set; } = null!;

    public string TenChuongTrinh { get; set; } = null!;

    public string Loai { get; set; } = null!;

    public decimal GiaTri { get; set; }

    public decimal? GiaTriToiDa { get; set; }

    public decimal DonHangToiThieu { get; set; }

    public int? SoLuongMa { get; set; }

    public int DaSuDung { get; set; }

    public DateTime NgayBatDau { get; set; }

    public DateTime NgayKetThuc { get; set; }

    public string TrangThai { get; set; } = null!;

    public virtual ICollection<DonHang> DonHangs { get; set; } = new List<DonHang>();

    public virtual ICollection<LichSuDungVoucher> LichSuDungVouchers { get; set; } = new List<LichSuDungVoucher>();
}
